util/io/paths.go

```diff
@@ -0,0 +1,37 @@
+package io
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+// TempPaths allows generating and memorizing random paths for a given URL.
+type TempPaths struct {
+	paths map[string]string
+	lock  sync.Mutex
+}
+
+func NewTempPaths() *TempPaths {
+	return &TempPaths{
+		paths: make(map[string]string),
+	}
+}
+
+// GetPath generates a path for the given URL or returns previously generated one.
+func (p *TempPaths) GetPath(url string) (string, error) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	if val, ok := p.paths[url]; ok {
+		return val, nil
+	}
+	uniqueId, err := uuid.NewRandom()
+	if err != nil {
+		return "", err
+	}
+	repoPath := filepath.Join(os.TempDir(), uniqueId.String())
```

Comment:
I would suggest using two uuids combined, because uuid will only be 16 characters in length.

Reply:
I think [one is probably fine](https://pkg.go.dev/github.com/google/uuid#NewRandom).

> Randomly generated UUIDs have 122 random bits.  One's annual risk of being
> hit by a meteorite is estimated to be one chance in 17 billion, that
> means the probability is about 0.00000000006 (6 × 10−11),
> equivalent to the odds of creating a few tens of trillions of UUIDs in a
> year and having one duplicate.

But if we really _need_ 122+ bits of entropy in the path, then we should change the `tempDir` creation logic in `ExtractChart`. It currently uses `ioutil.TempDir` which adds a (pseudo?) random uint32 suffix.

Reply:
On the other hand, if we don't need 122 bits, we could avoid error handling by using `itoa.Uitoa(uint(rand.Uint32()))`.

Reply:
I agree this is more than enough. @jannfis let me know please if you feel strongly we need two UUIDs. They are free, no harm to use two :)